src/utils/src/ExponentialBackoffRetryStrategy.c

```diff
@@ -205,18 +210,24 @@ STATUS validateAndUpdateExponentialBackoffStatus(PExponentialBackoffRetryStrateg
     return retStatus;
 }
 
-STATUS getExponentialBackoffRetryStrategyWaitTime(PRetryStrategy pRetryStrategy, PUINT64 retryWaitTime) {
+STATUS getExponentialBackoffRetryStrategyWaitTime(PKvsRetryStrategy pKvsRetryStrategy, PUINT64 retryWaitTime) {
     ENTERS();
     PExponentialBackoffRetryStrategyState pRetryState = NULL;
     PExponentialBackoffRetryStrategyConfig pRetryConfig = NULL;
     UINT64 currentSystemTime = 0, currentRetryWaitTime = 0, jitter = 0;
     STATUS retStatus = STATUS_SUCCESS;
 
-    if (pRetryStrategy == NULL || retryWaitTime == NULL) {
+    // CAUTION: DO NOT use CHK/CHK_STATUS macros here. In the code below, we're acquiring mutex which is getting
```

Comment:
Throughout the codebase, whenever we use MUTEX, we use a bool value `locked` and unlock only if `locked` is set to `TRUE`. Let us modify this to use that model in which case we avoid early returns. Check how `clientLocked` is used for example: https://github.com/awslabs/amazon-kinesis-video-streams-pic/blob/89a999684658da4e75e465dd601b4d56f4d91025/src/client/src/Client.c#L268

Reply:
Agree. Made the change.